server/src/test/java/com/google/speech/tools/voxetta/servlets/PromptServletTest.java

```diff
@@ -0,0 +1,123 @@
+package com.google.speech.tools.voxetta.servlets;
+
+import com.google.gson.Gson;
+import com.google.speech.tools.voxetta.data.Prompt;
+import com.google.speech.tools.voxetta.data.Prompt.Type;
+import com.google.speech.tools.voxetta.data.StatusResponse;
+import com.google.speech.tools.voxetta.services.DatastorePromptService;
+import com.google.speech.tools.voxetta.services.PromptService;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import jdk.net.SocketFlow.Status;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class PromptServletTest extends Mockito {
+
+
+    private HttpServletRequest request;
+    private HttpServletResponse response;
+    private PromptService promptService;
+    private PromptServlet servlet;
+
+    private final Gson gson = new Gson();
+
+    @Before
+    public void setUpMocks() {
+        request = Mockito.mock(HttpServletRequest.class);
+        response = Mockito.mock(HttpServletResponse.class);
+        promptService = Mockito.mock(DatastorePromptService.class);
+
+        servlet = new PromptServlet();
+        servlet.setService(promptService);
+    }
+
+    @Test
+    public void doGet_GetOnePrompt_ReturnsPrompt() throws IOException {
+
+        // mock prompt response
+        String mockedPrompt = gson.toJson(new Prompt(42312, Type.TEXT, "mocked body"));
+        when(promptService.getOnePrompt()).thenReturn(mockedPrompt);
+
+        // mock writer
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(printWriter);
+
+        // call doGet
+        servlet.doGet(request, response);
+
+        // verify response was called at least once
+        verify(response, atLeast(1)).setContentType("application/json");
+
+        // assert that prompt JSON is included in response
+        Assert.assertTrue(stringWriter.toString().contains(mockedPrompt));
+    }
+
+    @Test
+    public void doGet_GetOneEmptyPrompt_ReturnsEmptyPrompt() throws IOException {
+
+        // mock prompt response
+        String mockedPrompt = gson.toJson(new Object());
+        when(promptService.getOnePrompt()).thenReturn(mockedPrompt);
+
+        // mock writer
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(printWriter);
```

Comment:
Looks like lines 71-73 would be useful as a test utility function, stubStringWriter()

Reply:
stubbed